pygen/pygen_src/riscv_directed_instr_lib.py

```diff
@@ -0,0 +1,103 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+"""
+
+import vsc
+from enum import IntEnum, auto
+from pygen_src.riscv_instr_stream import riscv_rand_instr_stream
+from pygen_src.isa.riscv_instr import riscv_instr_ins
+from pygen_src.riscv_instr_gen_config import cfg
+from pygen_src.riscv_instr_pkg import riscv_reg_t
+from pygen_src.target.rv32i import riscv_core_setting as rcs
+from pygen_src.riscv_pseudo_instr import riscv_pseudo_instr
+
+
+class riscv_directed_instr_stream(riscv_rand_instr_stream):
+
+    def __init__(self):
+        super().__init__()
+        self.name = ""
+        self.label = ""
+
+    def post_randomize(self):
+        for i in range(len(self.instr_list)):
+            self.instr_list[i].has_label = 0
+            self.instr_list[i].atomic = 1
+        self.instr_list[0].comment = "Start %0s" % (self.name)
+        self.instr_list[-1].comment = "End %0s" % (self.name)
+        if self.label != "":
+            self.instr_list[0].label = self.label
+            self.instr_list[0].has_label = 1
+
+
+class int_numeric_e(IntEnum):
+    NormalValue = auto()
+    Zero = auto()
+    AllOne = auto()
+    NegativeMax = auto()
+
+
+@vsc.randobj
+class riscv_int_numeric_corner_stream(riscv_directed_instr_stream):
+    def __init__(self):
+        super().__init__()
+        self.num_of_avail_regs = vsc.uint8_t(10)
+        self.num_of_instr = vsc.rand_uint8_t()
+        self.init_val = vsc.rand_list_t(vsc.rand_bit_t(), sz = 10)
+        self.init_val_type = vsc.rand_list_t(vsc.enum_t(int_numeric_e), sz =10)
+        self.init_instr = []
+
+    """
+    TODO
+    @vsc.constraint
+    def init_val_c(self):
+        TO DO
+        solve init_val_type before init_val;
+        self.init_val_type.size in vsc.rangelist(self.num_of_avail_regs)
+        self.init_val.size in vsc.rangelist(self.num_of_avail_regs)
+        self.num_of_instr in vsc.rangelist(vsc.rng(15, 30))
+    @vsc.constraint
+    def avail_regs_c(self):
+        #self.dummy = self.avail_regs
+        vsc.unique(self.avail_regs)
+        with vsc.foreach(self.avail_regs, idx = True) as i:
+            self.avail_regs[i].not_inside(cfg.reserved_regs)
+            self.avail_regs[i] != 0
+    """
+
+    def pre_randomize(self):
+        # self.avail_regs = [None] * self.num_of_avail_regs
+        pass
+
+    def post_randomize(self):
+        self.init_instr = [None] * self.num_of_avail_regs
+        for i in range(len(self.init_val_type)):
+            if self.init_val_type[i] == int_numeric_e.Zero:
+                self.init_val[i] = 0
+            elif self.init_val_type[i] == int_numeric_e.AllOne:
+                self.init_val[i] = 1
+            elif self.init_val_type[i] == int_numeric_e.NegativeMax:
+                self.init_val[i] = 1 << (rcs.XLEN - 1)
+            self.init_instr[i] = riscv_pseudo_instr()
+            # TODO Issue with PyVSC.
```

Comment:
does this have a temporary workaround?

Reply:
Hi @udinator 
I am working on the temporary patch for this. I will update on this once done.

Reply:
Hi @udinator 
This is fixed now!